sdcm/nemesis.py

```diff
@@ -2398,6 +2399,165 @@ def disrupt_memory_stress(self):
         self.target_node.remoter.run(
             "stress-ng --vm-bytes $(awk '/MemTotal/{printf \"%d\\n\", $2 * 0.9;}' < /proc/meminfo)k --vm-keep -m 1 -t 100")
 
+    def disrupt_toggle_cdc_feature_properties_on_table(self):
+        """Manupulate cdc feature
+
+        randomly choose table and toggle cdc feature
+        with radomly generated cdc properties
+        """
+        self._set_current_disruption("ToggleCDCProperties")
+
+        if not self.tester.params.get("cdc_nemesis_enabled"):
+            self.log.warning("CDCStressor couldn't be run, executing cdc nemesis is disabled")
+            UnsupportedNemesis("CDCStressor couldn't be run, executing cdc nemesis is disabled")
+            return
+
+        def _generate_cdc_properties(cdc_enabled: bool) -> Tuple[str, str, str, int]:
+            """Generate random set of cdc properties
+
+            :param cdc_enabled: Was cdc enabled on table (True) or not (False)
+            :type cdc_enabled: bool
+            """
+            cdc_enabled = random.choice([True, False]) if cdc_enabled else True
+            preimage = random.choice([True, False]),
+            postimage = random.choice([True, False]),
+            ttl = random.randint(300, 3600)
+
+            return (cdc_enabled, preimage, postimage, ttl)
+
+        ks_tables_with_cdc = self.cluster.get_all_tables_with_cdc(self.target_node)
+        self.log.debug(f"Found next tables with enabled cdc feature: {ks_tables_with_cdc}")
+
+        if ks_tables_with_cdc:
+            ks, table = random.choice(ks_tables_with_cdc).split(".")
+            cdc_enabled, preimage, postimage, ttl = _generate_cdc_properties(cdc_enabled=True)
+        else:
+            ks_tables = self.cluster.get_non_system_ks_cf_list(self.target_node,
+                                                               filter_out_mv=True)
+            ks, table = random.choice(ks_tables).split(".")
+            cdc_enabled, preimage, postimage, ttl = _generate_cdc_properties(cdc_enabled=False)
+
+        self._alter_table_with_cdc_properties(ks, table, cdc_enabled, preimage, postimage, ttl)
+        self._verify_cdc_feature_status(ks, table, cdc_enabled, preimage, postimage, ttl)
+
+    def disrupt_run_cdcstressor_tool(self):
+        self._set_current_disruption(label="RunCDCStressorTool")
+
+        if not self.tester.params.get("cdc_nemesis_enabled"):
+            self.log.warning("CDCStressor couldn't be run, executing cdc nemesis is disabled")
+            UnsupportedNemesis("CDCStressor couldn't be run, executing cdc nemesis is disabled")
+            return
```

Comment:
Again, I don't think we need this param, we just need to check if the CDC is enabled and use it.

Reply:
Removed